src/ui/public/vis/map/service_settings.js

```diff
@@ -188,30 +132,26 @@ uiModules.get('kibana')
        * @param additionalQueryParams
        */
       addQueryParams(additionalQueryParams) {
-        for (const key in additionalQueryParams) {
-          if (additionalQueryParams.hasOwnProperty(key)) {
-            if (additionalQueryParams[key] !== this._queryParams[key]) {
-              //changes detected.
-              this._queryParams = _.assign({}, this._queryParams, additionalQueryParams);
-              this._invalidateSettings();
-              break;
-            }
-          }
-        }
+        this._emsClient.addQueryParams(additionalQueryParams);
       }
 
-      getEMSHotLink(fileLayer) {
-        const id = `file/${fileLayer.name}`;
-        return `${mapConfig.emsLandingPageUrl}#${id}`;
+      async getEMSHotLink(fileLayerConfig) {
```

Comment:
```suggestion
      getEMSHotLink(fileLayerConfig) {
```

Reply:
we need to preserver the asyncy-ness, since we use the ems-client to resolve the object. this is a hold-over from service_settings. the gis-app will just use the ems_client directly